velox/dwio/dwrf/test/TestReader.cpp

```diff
@@ -910,6 +910,32 @@ TEST(TestReader, testMismatchSchemaIncompatible) {
   }
 }
 
+TEST(TestReader, readAsLowerCase) {
+  // upper.orc holds one columns (Bool_Val: BOOLEAN, b: BIGINT)
+  ReaderOptions readerOpts{defaultPool.get()};
+  readerOpts.setReadAsLowerCase(true);
+  auto reader = DwrfReader::create(
+      createFileBufferedInput(
+          getExampleFilePath("upper.orc"), readerOpts.getMemoryPool()),
+      readerOpts);
+  auto type = reader->typeWithId();
+  auto col0 = type->childAt(0);
+  EXPECT_EQ(type->childByName("bool_val"), col0);
+}
+
+TEST(TestReader, readAsLowerCaseMap) {
+  // upper_map.orc holds one columns (mAp_val:map<int,float>)
```

Comment:
Map itself is ok.  The important ones are `map<int, struct<A: bigint>>`, `struct<A: struct<B: bigint>>`, `array<struct<A: bigint>>`.

Reply:
Do we have an example to generate dwrf format file with complex type? I find now I 'm hard to generate it. I generate the map file by `ColumnWriterStatsTest`

Reply:
Check the code here: https://github.com/facebookincubator/velox/blob/6fbdc2845810ad05fd693211f7560ee9c51d9999/velox/dwio/dwrf/test/E2EFilterTest.cpp#L76-L82

Reply:
Thanks! I have added test according to your guide.